File: Sol14_Longest_Common_Prefix/Solution14.java
import java.util.Arrays;

class Solution {
    public String longestCommonPrefix(String[] v) {
        Arrays.sort(v);

        String first = v[0];
        String last = v[v.length - 1];

        StringBuilder ans = new StringBuilder();

        for (int i = 0; i < Math.min(first.length(), last.length()); i++) {
            if (first.charAt(i) != last.charAt(i)) {
                break;
            }

            ans.append(first.charAt(i));
        }

        return ans.toString();
    }

    public static void main(String[] args) {
        String[] v = {"flower", "flow", "flight"};

        Solution solution = new Solution();
        System.out.println(solution.longestCommonPrefix(v));

        // Output:
        // fl
    }
}

File: Sol14_Longest_Common_Prefix/Solution14.py
from typing import List

class Solution:
    def longestCommonPrefix(self, v: List[str]) -> str:
        v.sort()

        first = v[0]
        last = v[-1]

        ans = ""

        for i in range(min(len(first), len(last))):
            if first[i] != last[i]:
                break

            ans += first[i]

        return ans


# Example
v = ["flower", "flow", "flight"]

solution = Solution()
print(solution.longestCommonPrefix(v))

# Output:
# fl